File: app/src/main/java/awais/instagrabber/models/enums/NotificationType.java
package awais.instagrabber.models.enums;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

public enum NotificationType implements Serializable {
    // web
    LIKE("GraphLikeAggregatedStory"),
    FOLLOW("GraphFollowAggregatedStory"),
    COMMENT("GraphCommentMediaStory"),
    MENTION("GraphMentionStory"),
    TAGGED("GraphUserTaggedStory"),
    // app story_type
    COMMENT_LIKE("13"),
    TAGGED_COMMENT("14"),
    RESPONDED_STORY("213"),
    // efr
    REQUEST("REQUEST");

    private final String itemType;
    private static final Map<String, NotificationType> map = new HashMap<>();

    static {
        for (NotificationType type : NotificationType.values()) {
            map.put(type.itemType, type);
        }
    }

    NotificationType(final String itemType) {
        this.itemType = itemType;
    }

    public String getItemType() {
        return itemType;
    }

    public static NotificationType valueOfType(final String itemType) {
        return map.get(itemType);
    }
}

File: app/src/main/java/awais/instagrabber/models/enums/NotificationType.kt
package awais.instagrabber.models.enums

import java.io.Serializable

enum class NotificationType(val itemType: Int) : Serializable {
    LIKE(60),
    FOLLOW(101),
    COMMENT(12),  // NOT TESTED
    COMMENT_MENTION(66),
    TAGGED(102),  // NOT TESTED
    COMMENT_LIKE(13),
    TAGGED_COMMENT(14),
    RESPONDED_STORY(213),
    REQUEST(75),
    AYML(9999);

    companion object {
        private val map: MutableMap<Int, NotificationType> = mutableMapOf()

        @JvmStatic
        fun valueOfType(itemType: Int): NotificationType? {
            return map[itemType]
        }

        init {
            for (type in values()) {
                map[type.itemType] = type
            }
        }
    }
}